dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/cache/DefaultCacheProvider.java

```diff
@@ -646,4 +647,15 @@ public <V> Cache<V> createSubExpressionCache()
             .forRegion( Region.subExpressionCache.name() )
             .expireAfterWrite( 5, TimeUnit.MINUTES ) );
     }
+
+    @Override
+    public <V> Cache<V> createQueryAliasCache()
+    {
+        return registerCache( this.<V> newBuilder()
+            .forRegion( Region.queryAliasCache.name() )
+            .expireAfterWrite( 3, TimeUnit.HOURS )
```

Comment:
Is there an intention with 3 hours specifically, or is it just an arbitrary number?

I'm initially thinking that this is not a service you would use over a long time, so 3 hours is not too long, but on the other hand, after our chat today, I was thinking this is a use once kind of service, so maybe we even drop it down to like 5 minutes or so? Just a though, not a request to change :)

Reply:
Yeah I think we could go shorter or longer, but probably don't need to worry too much about it unless this becomes problematic